### PR TITLE
fix: group should recursive search for child elements (#728)

### PR DESCRIPTION
**Pull request by hexianWeb**
**original func**
Mesh with events cannot be collected correctly when they wrapped by `TresGroup`, So when collections meet `Tresgroup`,  should recurse  to the child elements.

### CODE REVIEW

File: src/composables/useRaycaster/index.ts [hunk 1]
```diff
@@ -16,7 +16,16 @@ export const useRaycaster = (
...
+  const objectWihtEvents = computed(() => {
+    const hasEvents = (object)=> object.__tres?.eventCount > 0;
+    const checkObject = (object)=> {
+      if (object.isGroup) {
+        return object.children.some(checkObject)
+      }
+      return hasEvents(object)
+    }
+    return objects.value.filter(checkObject)
+  })
```
Comment (garrlker) [inline src/composables/useRaycaster/index.ts:28]: Hey @hexianWeb, thank you for the issue and the PR with a fix!

The solution looks good as is, but I'd like to clean it up a bit for readability

I know it's subjective on whether ternaries *are* more readable, so feel free to push back if you don't like this suggestion

```suggestion
  const hasEvents = (object) => object.__tres?.eventCount > 0;
  const isGroup = (object) => object.isGroup ? object.children.some(isGroup) : hasEvents(object)

  // Recurse through scene graph and filter for objects with events
  const objectWihtEvents = computed(() => objects.value.filter(isGroup))
```

Also, while this fixes the `Groups` case, any `Object3D` can contain children as well, so maybe the fix will look more like this?

```suggestion
  const hasEvents = (object) => object.__tres?.eventCount > 0;
  const hasChildren = (object) => object?.children.length > 0 ? object.children.some(hasChildren) : hasEvents(object)

  // Recurse through scene graph and filter for objects with events
  const objectWihtEvents = computed(() => objects.value.filter(hasChildren))
```

And 2 things that you didn't cause but would be nice if you could fix

1. Can we fix the `objectWihtEvents` typo? `objectWithEvents`
2. We're filtering for the events in the raycaster, but the source of these objects is coming from the EventManager. Can we move this logic into it's computed instead? [Link](https://github.com/Tresjs/tres/blob/main/src/composables/useTresEventManager/index.ts#L33)

**Review by garrlker — CHANGES_REQUESTED** [review]
Overall looks good! A couple suggestions and questions

File: src/composables/useRaycaster/index.ts (hunk 1 above)
Comment (hexianWeb) [reply src/composables/useRaycaster/index.ts]: Sorry for replying so late, I have changed the code. 
Please let me know if any code needs to be modified.

**Review by garrlker — APPROVED** [review]

**Comment by hexianWeb** [comment]
Merging is blocked! Is there anything I should do now?

File: src/composables/useRaycaster/index.ts [hunk 2]
```diff
@@ -9,14 +9,13 @@ import type { DomEvent, TresCamera, TresEvent } from 'src/types'
 import type { TresContext } from '../useTresContextProvider'
 
 export const useRaycaster = (
-  objects: Ref<Object3D[]>,
+  objectWithEvents: Ref<Object3D[]>,
```
Comment (andretchen0) [inline src/composables/useRaycaster/index.ts:12]: Can we please change this to `objectsWithEvents`? (The original was plural and the type is an array, so many `objects`.)

Thanks! And sorry for the late change request here.

**Review by andretchen0 — CHANGES_REQUESTED** [review]

File: src/composables/useRaycaster/index.ts (hunk 2 above)
Comment (hexianWeb) [reply src/composables/useRaycaster/index.ts]: Thank you for your prompt response

**Pull request merged**